services/src/main/java/org/keycloak/services/resources/IdentityBrokerService.java

```diff
@@ -796,9 +812,13 @@ private Response afterPostBrokerLoginFlowSuccess(AuthenticationSessionModel auth
             if (firstBrokerLoginInProgress) {
                 logger.debugf("Reauthenticated with broker '%s' when linking user '%s' with other broker", context.getIdpConfig().getAlias(), federatedUser.getUsername());
 
-                UserModel linkingUser = AbstractIdpAuthenticator.getExistingUser(session, realmModel, authSession);
-                if (!linkingUser.getId().equals(federatedUser.getId())) {
-                    return redirectToErrorPage(authSession, Response.Status.BAD_REQUEST, Messages.IDENTITY_PROVIDER_DIFFERENT_USER_MESSAGE, federatedUser.getUsername(), linkingUser.getUsername());
+                try {
+                    UserModel linkingUser = AbstractIdpAuthenticator.getExistingUser(session, realmModel, authSession);
```

Comment:
Instead of this change, could we remove the check for `if (!linkingUser.getId().equals(federatedUser.getId()))` entirely? With this PR, there is a way that user authenticated in username/password form could be different than the one in the "AbstractIdpAuthenticator.getExistingUser" . So I believe that same logic can be for broker re-authentication. This means that IDENTITY_PROVIDER_DIFFERENT_USER_MESSAGE can be removed entirely from the "Messages" class and resource bundles. I think the similar change may now need to be done in the ResetCredentialsActionTokenHandler

Reply:
done